### PR TITLE
Honor DOCKER_BUILDKIT in integration tests

### PR DESCRIPTION
**Pull request by tiborvass**
This PR refactors code in `integration/build/` and honors the DOCKER_BUILDKIT environment variable.

It introduces a new `integration/build/buildutil` package that could in the future be refactored into a easier to use docker client package.

All integration (non-cli) tests should pass with or without BuildKit enabled.

### CODE REVIEW

**Comment by tonistiigi** [comment]
@tiborvass CI errors

File: integration/build/util_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		if err != nil {
+			return filter, err
+		}
+		if ix := strings.Index(m.Stream, intermediateContainerPrefix); ix != -1 {
+			filter.Add("id", strings.TrimSpace(m.Stream[ix+len(intermediateContainerPrefix):]))
+		}
+	}
+}
+
+func writeTarRecord(t *testing.T, w *tar.Writer, fn, contents string) {
```
Comment (thaJeztah) [inline integration/build/util_test.go]: not introduced in this PR, but just noticed this should probably have `t.Helper()`?

File: integration/build/util_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+func withBuilderVersion(buildOptions types.ImageBuildOptions) types.ImageBuildOptions {
+	if os.Getenv("DOCKER_BUILDKIT") == "1" {
+		buildOptions.Version = types.BuilderBuildKit
+	}
+	return buildOptions
+}
+
+func init() {
+	defaultBuildOptions = withBuilderVersion(types.ImageBuildOptions{
```
Comment (vdemeester) [inline integration/build/util_test.go]: nit: why not

```go
var (
    defaultBuildOptions = withBuilderVersion(…)
)
```

**Review by vdemeester — APPROVED** [review]
LGTM with a small comment

**Comment by tiborvass** [comment]
Actually I have found an issue with this PR when buildkit is tested.

**Comment by thaJeztah** [comment]
```
04:10:00 Running /go/src/github.com/docker/docker/integration/build/buildutil
04:10:00 env: './test.main': No such file or directory
```


Also some linting issues;

```
04:03:28 integration/build/buildutil/util.go:84:1:warning: exported function Build should have comment or be unexported (golint)
04:03:28 integration/build/buildutil/util.go:78:6:warning: exported type BuildInput should have comment or be unexported (golint)
04:03:28 integration/build/buildutil/util.go:51:1:warning: exported method BuildResult.CacheHit should have comment or be unexported (golint)
04:03:28 integration/build/buildutil/util.go:39:1:warning: exported method BuildResult.OutputContains should have comment or be unexported (golint)
04:03:28 integration/build/buildutil/util.go:30:6:warning: exported type BuildResult should have comment or be unexported (golint)
04:03:28 integration/build/buildutil/util.go:26:1:warning: exported function BuildKitEnabled should have comment or be unexported (golint)
```

**Comment by tiborvass** [comment]
@thaJeztah fixed.

I moved `buildutil` to `builder/buildutil`.

**Comment by AkihiroSuda** [comment]
needs rebase

**Comment by AkihiroSuda** [comment]
ping^^ @tiborvass

**Comment by olljanat** [comment]
@tiborvass this one needs to be rebased

**Comment by thaJeztah** [comment]
carried in https://github.com/moby/moby/pull/38415

**Comment by olljanat** [comment]
Derek close

**Pull request closed without merging**